pkg/metrics/query.go

```diff
@@ -64,13 +65,33 @@ func Query(
 
 	var res model.Value
 
-	res, warn, err = a.Query(ctx, query.Query, time.Now())
-	if err != nil {
-		err = fmt.Errorf("querying: %w", err)
-		return warn, err
-	}
+	if query.Duration > 0 {
+		step := defaultStep
+		if query.Step > 0 {
+			step = query.Step
+		}
+
+		_, warn, err = a.QueryRange(ctx, query.Query, promapiv1.Range{
+			Start: time.Now().Add(-time.Duration(query.Duration)),
+			End:   time.Now(),
+			Step:  step,
+		})
+		if err != nil {
+			err = fmt.Errorf("querying: %w", err)
+			return warn, err
+		}
+
+		// Don't log response in range query case because there are a lot.
+		level.Debug(l).Log("msg", "request finished", "name", query.Name, "trace-id", rt.TraceID)
+	} else {
```

Comment:
nit: Maybe get rid of else and flatten out with a return.

Reply:
Done.